jujugui/static/gui/src/app/components/deployment/add-credentials/add-credentials.js

```diff
@@ -32,14 +35,39 @@ YUI.add('deployment-add-credentials', function() {
       @method _handleCloudClick
     */
     _handleAddCredentials: function(id) {
-      this.props.changeState({
-        sectionC: {
-          component: 'deploy',
-          metadata: {
-            activeComponent: 'summary'
+      // Add template
+      var user = this.props.users.jem.user,
+          templateName = this.refs.templateName.value,
+          template = {
+            // XXX The controllers need to exist first; for now use a known good
+            // controller with work to come to generate a controller name.  Also
+            // this key will change from 'state-server' to 'controller'.
+            // Makyo 2016-03-30
+            'state-server': 'yellow/aws-eu-central-1',
+            //'state-server': 'yellow/aws-' + this.refs.templateRegion.value,
+            // XXX This applies only to AWS (for first release).
+            'config': {
+              'access-key': this.refs.templateAccessKey.value,
+              'secret-key': this.refs.templateSecretKey.value
+            }
+          };
+      this.props.jem.addTemplate(
```

Comment:
This is probably fine for now but I don't think we want to actually add this template until they click deploy